test/speech/index.js

```diff
@@ -0,0 +1,836 @@
+
+    it('should make the correct API request', function(done) {
+      var request = REQ;
+
+      speech.request = function(protoOpts, reqOpts) {
+        assert.strictEqual(protoOpts.service, 'Speech');
+        assert.strictEqual(protoOpts.method, 'syncRecognize');
+        assert.strictEqual(reqOpts, REQ);
+
+        done();
+      };
+
+      speech.recognize(request, assert.ifError);
+    });
+
+    describe('error', function() {
+      var error = new Error('Error.');
+      var RES = {};
+
+      beforeEach(function() {
+        speech.request = function(protoOpts, reqOpts, callback) {
+          callback(error, RES);
+        };
+      });
+
+      it('should execute callback with error & API response', function(done) {
+        speech.recognize(REQ, function(err, response, apiResponse) {
+          assert.strictEqual(err, error);
+          assert.strictEqual(response, null);
+          assert.strictEqual(apiResponse, RES);
+          done();
+        });
+      });
+    });
+
+    describe('success', function() {
+      var RES = {
+        results: []
+      };
+
+      beforeEach(function() {
+        speech.request = function(protoOpts, reqOpts, callback) {
+          callback(null, RES);
+        };
+      });
+
+      it('should execute callback with results & API resp', function(done) {
+        speech.recognize(REQ, function(err, response, apiResponse) {
+          assert.ifError(err);
+
+          assert(response.results);
+          assert(response.get('results'));
+          assert.strictEqual(apiResponse, RES);
+
+          done();
+        });
+      });
+    });
+  });
+
+  describe('detect', function() {
+    var findFiles_;
+    var REQ = {};
+
+    var FILES = [
+      {
+        content: 'aGk='
+      }
+    ];
+
+    before(function() {
+      findFiles_ = Speech.findFiles_;
+    });
+
+    beforeEach(function() {
+      Speech.findFiles_ = function(files, callback) {
+        callback(null, FILES);
+      };
+    });
+
+    after(function() {
+      Speech.findFiles_ = findFiles_;
+    });
+
+    it('should find the files', function(done) {
+      Speech.findFiles_ = function(files) {
+        assert.strictEqual(files, FILE);
+        done();
+      };
+
+      speech.detect(FILE, REQ, assert.ifError);
+    });
+
+    it('should return an error from findFiles_', function(done) {
+      var error = new Error('Error.');
+
+      Speech.findFiles_ = function(files, callback) {
+        assert.strictEqual(files, FILE);
+        callback(error);
+      };
+
+      speech.detect(FILE, REQ, function(err) {
+        assert.strictEqual(err, error);
+        done();
+      });
+    });
+
+    it('should return empty detections when none were found', function(done) {
+      speech.recognize = function(REQ, callback) {
+        callback(null, new speech.protos.Speech.SyncRecognizeResponse({
+          results: []
+        }));
+      };
+
+      speech.detect(
+        FILE,
+        REQ,
+        function(err, response) {
```

Comment:
We should assert that the `apiResponse` is present and also that `response` is the `SyncRecognizeResponse` instance.